internal/column-helpers/columnhelpers.go

```diff
@@ -0,0 +1,307 @@
+		if enrichers != nil {
+			for _, enricher := range enrichers {
+				for _, ev := range events {
+					enricher(ev)
+				}
+			}
+		}
+		ch.snapshotCombiner.AddSnapshot(key, events)
+	}
+}
+
+func (ch *ColumnHelpers[T]) writeErrorMessage(severity int, message string) {
+	if ch.errorCallback == nil {
+		return
+	}
+	ch.errorCallback(severity, message)
+}
+
+func (ch *ColumnHelpers[T]) JSONHandlerFunc(enrichers ...func(any)) func([]byte) {
+	if ch.eventCallback == nil {
+		panic("set event callback before getting the EventHandlerFunc from ColumnHelper")
+	}
+	handler := ch.eventHandler(enrichers...)
+	return func(event []byte) {
+		ev := new(T)
+		err := json.Unmarshal(event, ev)
+		if err != nil {
+			ch.writeErrorMessage(0, fmt.Sprintf("Error unmarshalling: %v", err))
+			return
+		}
+		handler(ev)
+	}
+}
+
+func (ch *ColumnHelpers[T]) JSONHandlerFuncArray(enrichers ...func(any)) func([]byte) {
+	if ch.eventCallback == nil {
+		panic("set event callback before getting the EventHandlerFunc from ColumnHelper")
+	}
+	handler := ch.eventHandlerArray(enrichers...)
+	return func(event []byte) {
+		var ev []*T
+		err := json.Unmarshal(event, &ev)
+		if err != nil {
+			ch.writeErrorMessage(0, fmt.Sprintf("Error unmarshalling: %v", err))
+			return
+		}
+		handler(ev)
+	}
+}
+
+func (ch *ColumnHelpers[T]) EventHandlerFunc(enrichers ...func(any)) any {
+	if ch.eventCallback == nil {
+		panic("set event callback before getting the EventHandlerFunc from ColumnHelper")
+	}
+	return ch.eventHandler(enrichers...)
+}
+
+func (ch *ColumnHelpers[T]) EventHandlerFuncArray(enrichers ...func(any)) any {
+	if ch.eventCallbackArray == nil {
+		panic("set event callback before getting the EventHandlerFunc from ColumnHelper")
+	}
+	return ch.eventHandlerArray(enrichers...)
+}
+
+func (ch *ColumnHelpers[T]) EventHandlerFuncSnapshot(key string, enrichers ...func(any)) any {
+	if ch.eventCallbackArray == nil {
+		panic("set event callback before getting the EventHandlerFunc from ColumnHelper")
+	}
+	return ch.eventHandlerSnapshot(key, enrichers...)
+}
+
+func (ch *ColumnHelpers[T]) GetTextColumnsFormatter(options ...textcolumns.Option) TextColumnsFormatter {
+	return &outputHelper[T]{
+		ch:                   ch,
+		TextColumnsFormatter: textcolumns.NewFormatter(ch.columns.GetColumnMap(ch.columnFilters...), options...),
+	}
+}
+
+func (ch *ColumnHelpers[T]) GetColumnNamesAndDescription() map[string]string {
+	out := make(map[string]string)
+	for _, column := range ch.columns.GetOrderedColumns(ch.columnFilters...) {
+		out[column.Name] = column.Description
+	}
+	return out
+}
+
+func (ch *ColumnHelpers[T]) GetDefaultColumns() []string {
+	cols := make([]string, 0)
+	for _, column := range ch.columns.GetOrderedColumns(ch.columnFilters...) {
+		if !column.Visible {
+			continue
+		}
+		cols = append(cols, column.Name)
+	}
+	return cols
+}
+
+func (ch *ColumnHelpers[T]) SetSorting(sortBy []string) error {
+	ch.columns.VerifyColumnNames(sortBy)
+	ch.sortBy = sortBy
```

Comment:
I currently don't know how hot this path is. But maybe we could do a `columns.Sort.Prepare` here?

Reply:
Yes! It's actually completely missing the sorting right now as it seems. Well, I'm lagging behind with the unit tests... :/